src/components/calculator.js

```diff
@@ -1,42 +1,61 @@
 import React from 'react';
+import calculate from '../logic/calculate';
+
+class Calculator extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: 0,
+      next: 0,
+    };
+    this.handleClick = this.handleClick.bind(this);
+    this.handleDisplay = this.handleDisplay.bind(this);
+  }
+
+  handleClick(event) {
+    const result = calculate(this.state, event.target.value);
+    this.setState(result);
+  }
+
+  handleDisplay(event) {
+    this.setState({
+      total: event.target.value,
+    });
+  }
 
-// eslint-disable-next-line react/prefer-stateless-function
-class Calculator extends React.Component {
-  // constructor(props) {
-  //   super(props);
-  // }
   render() {
+    const currentState = this.state;
     return (
       <div className="calculator">
-        <div className="screen">0</div>
+        <div className="screen" onChange={this.handleDisplay}>{currentState.total || currentState.next || 0}</div>
         <div className="row">
```

Comment:
- [ ] When using your calculator app, I notice that only the initial value clicked and the result of the operation gets displayed. This makes it difficult to know if the buttons are pressing or even if the app is working well at all.
If you are an addition operation, say `2 + 2`, let the display show that 🙏

Reply:
Thank you @codepantha 🙏